_posts/2015-04-24-FFI-and-Rust.md

```diff
@@ -0,0 +1,264 @@
+---
+layout: post
+title: "FFI and Rust"
+author: Alex Crichton
+description: "Zero-cost and safe FFI in Rust"
+---
+
+
+Rust's quest for world domination was never destined to happen overnight, so
+Rust needs to be able to interoperate with the the existing world just as easily
+as it talks to itself. To solve this problem, **Rust lets you communicate with C
+APIs at no extra cost while providing strong safety guarantees**.
+
+This is also referred to as Rust's foreign function interface (FFI) and is the
+method by which Rust communicates with other programming languages. In following
+Rust's design principles, this is a **zero cost abstraction** which leverages
+language features such as ownership and borrowing to provide a **safe
+interface**.
```

Comment:
Hm I was trying to get both points up here to start emphasizing them both right off the bat, but I do agree that the wording here is a little odd. I'll see what I can do about this.